applications/trilinos_application/python_scripts/trilinos_import_model_part_utility.py

```diff
@@ -40,26 +40,46 @@ def ImportModelPart(self):
-                if KratosMPI.mpi.rank == 0:
-
-                    # Original .mdpa file reading
-                    model_part_io = KratosMultiphysics.ReorderConsecutiveModelPartIO(input_filename)
-
-                    # Partition of the original .mdpa file
-                    number_of_partitions = KratosMPI.mpi.size # Number of partitions equals the number of processors
-                    domain_size = self.main_model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE]
-                    verbosity = self.settings["echo_level"].GetInt()
-                    sync_conditions = True # Make sure that the condition goes to the same partition as the element is a face of
-                    partitioner = KratosMetis.MetisDivideHeterogeneousInputProcess(model_part_io, number_of_partitions , domain_size, verbosity, sync_conditions)
+                
+                # Partition of the original .mdpa file
+                number_of_partitions = KratosMPI.mpi.size # Number of partitions equals the number of processors
+                domain_size = self.main_model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE]
+                verbosity = self.settings["echo_level"].GetInt()
+
+                # Make sure that the condition goes to the same partition as the element is a face of
+                sync_conditions = True
+
+                # Original .mdpa file reading
+                model_part_io = KratosMultiphysics.ReorderConsecutiveModelPartIO(input_filename)
+
+                if perform_partitioning_in_memory == "File":
+                    ## Serial partition of the original .mdpa file
+                    if KratosMPI.mpi.rank == 0:
+                        partitioner = KratosMetis.MetisDivideHeterogeneousInputProcess(model_part_io, number_of_partitions , domain_size, verbosity, sync_conditions)
+                        partitioner.Execute()
+
+                        KratosMultiphysics.Logger.PrintInfo("::[TrilinosImportModelPartUtility]::", "Metis divide finished.")
+                elif perform_partitioning_in_memory == "Memory":
+                    # Create a second io that does not reorder the parts while reading from memory
+                    serial_model_part_io = KratosMultiphysics.ModelPartIO(input_filename)
+
+                    partitioner = KratosMetis.MetisDivideHeterogeneousInputInMemoryProcess(model_part_io, serial_model_part_io, number_of_partitions , domain_size, verbosity, sync_conditions)
                     partitioner.Execute()
+                    serial_model_part_io.ReadModelPart(self.main_model_part)
 
-                    KratosMultiphysics.Logger.PrintInfo("::[TrilinosImportModelPartUtility]::", "Metis divide finished.")
+                    if KratosMPI.mpi.rank == 0:
+                        KratosMultiphysics.Logger.PrintInfo("::[TrilinosImportModelPartUtility]::", "Metis divide finished.")
+                else:
+                    raise Exception("Invalid reading method selected") 
```

Comment:
`raise Exception('Invalid "partitioning_method" selected, possibilities are "file" and "memory"')`

Reply:
:+1: